tests/test_regrid.py

```diff
@@ -822,24 +1056,43 @@ def test_grid_raises_error_when_dataset_has_multiple_dims_for_an_axis(self):
         with pytest.raises(ValueError):
             ds_bounds.regridder.grid
 
-    def test_valid_tool(self):
+    def test_horizontal_tool_check(self):
         mock_regridder = mock.MagicMock()
         mock_regridder.return_value.horizontal.return_value = "output data"
 
         mock_data = mock.MagicMock()
 
-        with mock.patch.dict(accessor.REGRID_TOOLS, {"regrid2": mock_regridder}):
-            output = self.ac.horizontal("ts", mock_data, "regrid2")
+        with mock.patch.dict(
+            accessor.HORIZONTAL_REGRID_TOOLS, {"regrid2": mock_regridder}
+        ):
+            output = self.ac.horizontal("ts", mock_data, tool="regrid2")
 
         assert output == "output data"
 
         mock_regridder.return_value.horizontal.assert_called_with("ts", self.data)
 
-    def test_invalid_tool(self):
         with pytest.raises(
-            ValueError, match=r"Tool 'test' does not exist, valid choices"
+            ValueError, match=r"Tool 'dummy' does not exist, valid choices"
+        ):
+            self.ac.horizontal("ts", mock_data, tool="dummy")  # type: ignore
+
+    def test_vertical_tool_check(self):
+        mock_regridder = mock.MagicMock()
+        mock_regridder.return_value.vertical.return_value = "output data"
+
+        mock_data = mock.MagicMock()
+
+        with mock.patch.dict(accessor.VERTICAL_REGRID_TOOLS, {"xgcm": mock_regridder}):
+            output = self.ac.vertical("ts", mock_data, tool="xgcm", target_data=None)
+
+        assert output == "output data"
```

Comment:
The use of MagicMock is neat here. For those interested (probably just me), the xgcm vertical regridder object is being mocked (`mock_regridder`) and set as the `xgcm` dictionary value in `accessor.VERTICAL_REGRID_TOOLS`. The return value of `mock_regridder` is set to `"output_data"`, which is being tested in the assertion on line 1088. This test is to check if the correct vertical regridding tool is being referenced. 

The actual result of the regridder object is not being tested (hence `"output_data"`) because it is already tested through `class TestXGCMRegridder`.